pkg/channel/message_dispatcher.go

```diff
@@ -291,16 +294,28 @@ func (d *MessageDispatcherImpl) dispatchExecutionInfoTransformers(destination *u
 	if destination == nil {
 		destination = &url.URL{}
 	}
+
+	httpResponseBody := dispatchExecutionInfo.ResponseBody
+	if strings.Contains(destination.Host, "broker-filter") {
+		var errExtensionInfo broker.ErrExtensionInfo
+
+		err := json.Unmarshal(dispatchExecutionInfo.ResponseBody, &errExtensionInfo)
+		if err != nil {
+			d.logger.Debug("Unmarshal dispatchExecutionInfo ResponseBody failed", zap.Error(err))
+		}
+		destination = errExtensionInfo.ErrDestination
+		httpResponseBody = errExtensionInfo.ErrResponseBody
```

Comment:
If err != nil, we will endup dereferencing nil pointers at line 306-307

Reply:
ok understood